text/go/0001-go-bom.md

```diff
@@ -0,0 +1,163 @@
+installed via `go install github.com/CycloneDX/cyclonedx-gomod`.
+
+This tool supports three different cases of BOM generation:
+
+1. `app`: Include only those modules that the target application actually depends on. Modules required by tests or
+   packages that are not imported by the application are not included. Build constraints are evaluated, which enables a
+   very detailed view of what's really compiled into an application's binary.
+2. `mod`: Include the aggregate of modules required by all packages in the target module. This optionally includes
+   modules required by tests and test packages. Build constraints are NOT evaluated, allowing for a "whole picture" view
+   on the target module's dependencies.
+3. `bin`: Offers support of generating rudimentary SBOMs from binaries built with Go modules.
+
+Of these, we are only interested in `app`. The buildpacks do not currently support building modules, or libraries, which
+is the primary use-case for `mod`. The `bin` command is for pre-built executables, which is unnecessary for the
+buildpack as it has access to the source code. The BOM created by the `bin` command also contains less information than
+the one created via the `app` command.
+
+The output BOM contains the following fields for each module:
+
+* name,
+* version,
+* hash,
+* source URI,
+* license ID,
+* and package URL
+
+Note: The `license` field can be retrieved when the flag `-licenses` is passed to the command.
+
+Almost all of fields that we aim to support
+per [the Paketo BOM RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/0033-bill-of-materials.md#overall-schema)
+are available. The only one that is not available is `description`, and this is because there is no standard way to
+capture this information in Go projects (as compared with other languages like nodejs and rust, where the package
+description files contain a field for this metadata). See unresolved questions below.
+
+In order to support this tool in both the online and offline cases, we will host the binary tool on
+the [dep-server](https://github.com/paketo-buildpacks/dep-server).
+
+### Performance
+
+The addition of this tool to the language family order groupings might have an impactful change to the overall
+performance of the buildpacks.
+
+Performance metrics are still TBD, but for a typical project (e.g. `pack`) we observe that it takes about 10-30 seconds
+to construct a BOM.
+
+We can investigate ways to improve this performance.
+
+### Language Family Additions
+
+Using the CycloneDX tool to generate the BOM will involve some changes to the Go language family buildpack order groups
+as follows:
+
+* A single BOM generation buildpack will be added. It's primary functions are:
+    * perform the BOM dependency tool retrieval from the dep-server,
+    * execute BOM generation with the tool (`cyclonedx-gomod app -o bom.json`), and
+    * contribute the BOM to the `launch.toml` `[bom]` label.
```

Comment:
Might be worth noting that an extra conversion step will need to be performed between these two steps to convert the output JSON file to the legacy CNB TOML label format.

Reply:
Sure, we can add that.